src/middleware/with-logger.ts

```diff
@@ -0,0 +1,26 @@
+import { Middleware } from "./types.ts"
+
+export type Logger = {
+  debug: (...args: any[]) => void
+  info: (...args: any[]) => void
+  warn: (...args: any[]) => void
+  error: (...args: any[]) => void
```

Comment:
`trace`, `fatal`?

Reply:
I think these are fine to get started with, we can add those if needed later. Not sure if most loggers have `.trace` and `.fatal`.

Reply:
aight